compiler/noirc_frontend/src/node_interner.rs

```diff
@@ -559,7 +560,7 @@ impl NodeInterner {
                 // This lets us record how many arguments the type expects so that other types
                 // can refer to it with generic arguments before the generic parameters themselves
                 // are resolved.
-                TypeVariable::unbound(TypeVariableId(0))
+                ResolvedGeneric::dummy()
```

Comment:
Can you make an issue for removing these here as well?

If we require `push_empty_trait` and other def collection methods to be passed generics as well we shouldn't need these dummy methods or the old `TypeVariableId(0)` anymore.

Reply:
I ended up just removing the usage of `ResolvedGeneric::dummy` entirely with the move to resolving generics during def collection.